operator/v1/types_csi_cluster_driver.go

```diff
@@ -168,6 +168,63 @@ type AWSCSIDriverConfigSpec struct {
 	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov|aws-iso|aws-iso-b|aws-iso-e|aws-iso-f):kms:[a-z0-9-]+:[0-9]{12}:(key|alias)\/.*$`
 	// +optional
 	KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+
+	// efsVolumeMetrics sets the configuration for collecting metrics from EFS volumes used by the EFS CSI Driver.
+	// +openshift:enable:FeatureGate=AWSEFSDriverVolumeMetrics
+	EFSVolumeMetrics AWSEFSVolumeMetrics `json:"efsVolumeMetrics"`
+}
+
+// AWSEFSVolumeMetricsState defines the modes for collecting volume metrics in the AWS EFS CSI Driver.
+// This can either enable recursive collection of volume metrics or disable metric collection entirely.
+// +openshift:enable:FeatureGate=AWSEFSDriverVolumeMetrics
+// +kubebuilder:validation:Enum:="RecursiveWalk";"Disabled"
+type AWSEFSVolumeMetricsState string
+
+const (
+	// AWSEFSVolumeMetricsRecursiveWalk indicates that volume metrics collection in the AWS EFS CSI Driver
+	// is performed by recursively walking through the files in the volume.
+	AWSEFSVolumeMetricsRecursiveWalk AWSEFSVolumeMetricsState = "RecursiveWalk"
+
+	// AWSEFSVolumeMetricsDisabled indicates that volume metrics collection in the AWS EFS CSI Driver is disabled.
+	AWSEFSVolumeMetricsDisabled AWSEFSVolumeMetricsState = "Disabled"
+)
+
+// AWSEFSVolumeMetrics defines the configuration for volume metrics in the EFS CSI Driver.
+// +union
+type AWSEFSVolumeMetrics struct {
+	// state defines the state of metric collection in the AWS EFS CSI Driver. Enabling this option
+	// will cause the AWS EFS CSI Driver to recursively scan volumes to collect metrics. This process
+	// may result in high CPU and memory usage, depending on the volume size.
+	// The default value is Disabled.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default:=Disabled
+	// +default="Disabled"
+	State AWSEFSVolumeMetricsState `json:"state,omitempty"`
+
+	// options provides additional configuration for volume metrics in the AWS EFS CSI Driver.
+	// +unionMember,optional
+	Options *AWSEFSVolumeMetricsOptions `json:"options,omitempty"`
```

Comment:
Union members should be tied to a specific value of the discriminator, so this field should be `RecursiveWalk`. If the options are generic and apply to other versions later we can duplicate them within their respective structs

Reply:
Sorry, I'm not sure I understand this one. Do you mean that I should replace `Options` with `RecursiveWalk` (and potentially replace `AWSEFSVolumeMetricsOptions` with something more representative, like `AWSEFSVolumeMetricsRecursiveWalkOptions`)?